Create maintainers-minutes-2025-01-31.mdx (#3025)
nf-core/website: sites/main-site/src/content/blog/2025/maintainers-minutes-2025-01-31.mdx

headerImageDim: [250, 250]
embedHeaderImage: true
authors:
    - 'maxulysse'
label:
    - 'maintainers'
---

import Profile from '@components/GitHubProfilePictureExtended.svelte';
import { Image } from 'astro:assets';

The 'Maintainers Minutes' aims to give further insight into the workings of the [nf-core maintainers team](/governance#maintainers)
by providing brief summaries of the monthly team meetings.

## Overview

In the January meeting we discussed (amongst others) the following topics:

-   [Spring cleaning](#spring-cleaning)
-   [References](#references)
-   [stubs](#stubs)
-   [CI with latest-everything](#ci-with-latest-everything)
-   [Helpdesk Americas](#helpdesk-americas)
-   [Default prefixes in modules](#default-prefixes-in-modules)
-   [Best practices for spring cleaning](#best-practices-for-spring-cleaning)

## Spring cleaning

We decided that the spring cleaning that we planned to do before the hackathon should be done two weeks before the hackathon.
So it will happen from March 10th to March 14th.
We decided to extend the duration of the spring cleaning to a full week to give everyone enough time to participate.

> The idea is to clean up issues, PRs, have a look at the repos and open new issues if needed.

Reviewer: Maybe a expand with a bit more datail, the community might be interested.

E.g. stuff like: we go through common repos, and close PR that are older than X years (After giving a warning), deleting merged branches, we check slack/github teams to make sure they are up to date, archiving pipelines

## References

> Maxime Garcia wanted to know about the usage of `conf/igenomes.config`, and how it was actually used in the pipelines.

Reviewer: People do use local cloens of igenomes at least... but I thin ktaht config is the remote onee right?

Author: That config is the one people should be using and modifying, I will not move or update any file in igenomes, just adding new ones, and providing an easier way to access these files

Since no one is really actually using it, he will move forward with his ideas, and will continue to work on his current POC.

> Whenever Maxime says “I can do whatever I want” I get really scared.

## Stubs

> The discussion about stubs was going a bit in circles and off topic.

Reviewer: What was the discussion about? Have a leading sentence or a bette rtitle

It all started because the Nextflow language server was complaining about unused args in the stub section of any modules.
So people started to remove them, but then we have a stub block that is widely different from the script block, and no really way of knowing if the args are passed correctly within stubs.

We agreed that for now:

-   We will echo args for now instead of removing them
-   Talk with Ben, in particular in the static type PR and see if we can be smarter about it

## CI with latest-everything

> We widely agreed that since `latest-everything` can break a pipeline, it is important to test with it, because we want our code to be up to date with the latest update.

Reviewer: Why is it breaking? Give more context

But we also agreed that we don't want failure in this tests to block PRs.

> The end idea, is we keep the tests, but we don't fail on them.

Reviewer 2: ```suggestion
The end idea is that we keep the `latest-everything` tests in the CI workflow, but we don't require them to merge the PR.
```

## Helpdesk Americas

We discussed the helpdesk in the Americas timezone.
Since the start of the year, Florian and Lorena have mainly been on their own with very little audience.
We thought about ideas to improve it, and came up with the following:

> - Maybe change time, and delay it a bit

Reviewer 2: ```suggestion
- Consider moving the helpdesk to a later time slot so that it is outside of the busiest hours  
```

> -   Maybe change time, and delay it a bit

Author: ```suggestion
-   Consider moving the helpdesk to a later time slot so that it is outside of the busiest hours.
```

- Pump up on the communication, and make it more visible
  - Explaining clearly what the helpdesk is, and that there are clearly no stupid questions
  - See with the outreach team if we can get more visibility

Since we have all our sessions planed ahead, we can maybe communicate about it in a more efficient manner, as we currently rely on the people hosting the helpdesk to communicate about it.

## Default prefixes in modules

Our amazing modules aficionado Simon kindly reminded us about [this PR](https://github.com/nf-core/website/pull/2608).
It was decided that four maintainers should review it, so that we can hopefully close the discussion and advance on this topic and merge the PR.

## Best practices for spring cleaning

> We discussed Spring Cleaning once more, and our relentless modules dark nightSimon shared a nice idea:

Reviewer 2: ```suggestion
We discussed Spring Cleaning once more, and our relentless modules dark knight Simon shared a nice idea:
```

> > We should check our open PRs and close them if they are no longer relevant.

Reviewer: ```suggestion
> We should check our own open PRs and close them if they are no longer relevant.
```

It would be nice indeed to keep track of all our open PRs, and issues and close them if we can.

> That would simplify the spring cleaning, and all of our Justice League of maintainers own lives.

Reviewer 2: ```suggestion
That would simplify the spring cleaning, and all of our Justice League of maintainers' own lives.
```
